src/implementations/lq.jl

```diff
@@ -181,7 +181,7 @@ function lq_via_qr!(A::AbstractMatrix, L::AbstractMatrix, Q::AbstractMatrix,
                     qr_alg::AbstractAlgorithm)
     m, n = size(A)
     minmn = min(m, n)
-    At = adjoint!(similar(A'), A)
+    At = adjoint!(similar(A'), A)::AbstractMatrix
```

Comment:
Aha, this is where the type assert was added. Is this only JET that complains, or was there really a type instability or type inference issue? I cannot see how asserting an abstract type would help with anything.